edk2toolext/edk2_invocable.py

```diff
@@ -422,10 +425,14 @@ def ParseCommandLineOptions(self):
         BuildConfig = os.path.abspath(args.build_config)
 
         for argument in unknown_args:
-            if argument.count("=") != 1:
-                raise RuntimeError(f"Unknown variable passed in via CLI: {argument}")
-            tokens = argument.strip().split("=")
-            env.SetValue(tokens[0].strip().upper(), tokens[1].strip(), "From CmdLine")
+            if argument.count("=") == 1:
+                tokens = argument.strip().split("=")
+                env.SetValue(tokens[0].strip().upper(), tokens[1].strip(), "From CmdLine")
+                continue
+            elif argument.count("=") == 0:
+                env.SetValue(argument.strip().upper(), "TRUE", "From CmdLine")
```

Comment:
as we discussed this should be changed to either something totally random or some sort of string that indicates this value can not be used for value passing and should only be used for ifdef/ifndef like directives.

Reply:
@spbrogan I have updated the value of the variable to be a random string, and also updates the debug comment to state "Non Valued Variable Set From CmdLine"

I'm assuming ifdef/ifndef, you mean something along the lines of:

```python

if env.GetValue("NON_VALUED_VAR"):
  print("It exists")

if not env.GetValue("NON_VALUED_VAR"):
  print("It does not exist")
```

Then yes, this meets your expectations. Let me know!